### DISCUSSION TITLE
Plan to Drop Support for Qt 5 and Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by RytoEX** [obsproject/obs-studio#9055] [Announcements]
# Overview

With the one year anniversary of the completion of the [previously announced switch to Qt 6](https://github.com/obsproject/obs-studio/discussions/6481), we intend to remove support for Qt 5 and Qt 6 versions prior to 6.2.

As a result, we will also remove support for Ubuntu 20.04 which does not provide Qt 6 libraries.

In tandem with removing support for Ubuntu 20.04, we will also bump the minimum supported versions for other third-party libraries such as FFmpeg to the versions provided by Ubuntu 22.04.

# Why Drop Qt 5 and Ubuntu 20.04?

As we continue to work on improvements to the UI/UX side of OBS, it becomes increasingly burdensome to support Qt 5. 
Currently, the only first-party distribution of OBS using Qt 5 is Ubuntu 20.04, which has a dwindling user base in comparison to all other variants.

Additionally, having to support older dependency versions such as FFmpeg on Ubuntu 20.04 has become similarly tiresome.

# When Will Things Happen?

## Timetable

| Timeframe         | Action                                                                             |
| ----------------- | ---------------------------------------------------------------------------------- |
| Starting Now      | Removal of Qt 5 deps builds, ifdef'd code for Qt < 6.2, and Ubuntu 20.04 CI builds |
| Starting June     | Merge of new Qt 6.2+ only code (custom widgets, refactorings, etc.)                |
| Later             | Stable release of next major OBS version                                           |

## Starting Now

We will immediately begin work to:
1. stop building Qt 5 on our [obs-deps](https://github.com/obsproject/obs-deps) repository
2. remove code in [obs-studio](https://github.com/obsproject/obs-studio), [obs-browser](https://github.com/obsproject/obs-browser), [obs-websocket](https://github.com/obsproject/obs-websocket), and [obs-plugintemplate](https://github.com/obsproject/obs-plugintemplate) that enables building against Qt 5
3. remove our Ubuntu 20.04 CI builds on [obs-studio](https://github.com/obsproject/obs-studio)

## Starting June

We will begin merging new feature PRs that will require Qt 6.2+.

## Later

At a future point in time, we will release the next major version of OBS that will only support Qt 6.

# How Does This Affect Me?

## Users

Users of the first-party OBS distributions on Windows, macOS, Flatpak, or Ubuntu 22.04 will not see any impact. Ubuntu 20.04 users who wish to continue receiving new major/minor releases of OBS are encouraged to upgrade to Ubuntu 22.04 or later, or switch to using the Flatpak version of OBS.

## Package maintainers

We recognize that this change may impact third-party Linux distributions which currently use Qt 5. We encourage those to move to Qt 6 along with us. If that is not possible, such as on stable distributions not providing Qt 6 libraries, maintainers can stick to the `release/29.1` branch which will receive bug fixes at least until the release of the next major OBS version.

# Future considerations

We currently intend to support Qt 6.2 and Ubuntu 22.04 at least until the first major OBS version published 12 months after the next Ubuntu LTS version release (24.04). Ubuntu 24.04 is expected to be released in April 2024, and a year after that would be April 2025. This means that the earliest we would drop support for Qt 6.2 and Ubuntu 22.04 would be the first major version of OBS after April 2025. This is subject to change, but we will provide another announcement like this if our plans should change.

**Comment by Fenrirthviti** [comment]
This work has been complete, and Qt5 is no longer supported.
